demo/dia_dialogue_demo/run.py

```diff
@@ -0,0 +1,108 @@
+import gradio as gr
+import httpx
+
+
+emotions = [
+    "(laughs)",
+    "(clears throat)",
+    "(sighs)",
+    "(gasps)",
+    "(coughs)",
+    "(singing)",
+    "(sings)",
+    "(mumbles)",
+    "(beep)",
+    "(groans)",
+    "(sniffs)",
+    "(claps)",
+    "(screams)",
+    "(inhales)",
+    "(exhales)",
+    "(applause)",
+    "(burps)",
+    "(humming)",
+    "(sneezes)",
+    "(chuckle)",
+    "(whistles)",
+]
+speakers = ["Speaker 1", "Speaker 2"]
+
+client = httpx.AsyncClient(timeout=180)
+API_URL = "https://router.huggingface.co/fal-ai/fal-ai/dia-tts"
+
+
+async def query(dialogue: str, token: gr.OAuthToken | None):
+    if token is None:
+        raise gr.Error(
+            "No token provided. Use Sign in with Hugging Face to get a token."
+        )
+    headers = {
+        "Authorization": f"Bearer {token.token}",
+    }
+    response = await client.post(API_URL, headers=headers, json={"text": dialogue})
+    url = response.json()["audio"]["url"]
+    print("URL: ", url)
+    return url
+
+
+def formatter(speaker, text):
+    speaker = speaker.split(" ")[1]
+    return f"[S{speaker}] {text}"
+
+
+with gr.Blocks() as demo:
+    with gr.Sidebar():
+        login_button = gr.LoginButton()
+    gr.HTML(
+        """
+        <h1 style='text-align: center; display: flex; align-items: center; justify-content: center;'>
+        <img src="https://huggingface.co/datasets/freddyaboulton/bucket/resolve/main/dancing_huggy.gif" alt="Dancing Huggy" style="height: 100px; margin-right: 10px"> Dia Dialogue Generation Model
+        </h1>
+        <h2 style='text-align: center; display: flex; align-items: center; justify-content: center;'>Model by <a href="https://huggingface.co/nari-labs/Dia-1.6B"> Nari Labs</a>. Powered by HF and <a href="https://fal.ai/">Fal AI</a>  API.</h2>
```

Comment:
For some reason I'm not seeing spaces before and after the links:

<img width="662" alt="image" src="https://github.com/user-attachments/assets/12ffbfc6-df83-4db0-a708-19291953dca8" />


```suggestion
        <h2 style='text-align: center; display: flex; align-items: center; justify-content: center;'>Model by &nbsp;<a href="https://huggingface.co/nari-labs/Dia-1.6B"> Nari Labs</a>. Powered by HF and &nbsp; <a href="https://fal.ai/">Fal AI</a>&nbsp;  API.</h2>
```

Reply:
Really great demo! Btw for some reason I'm getting some strange outputs. The output audio is always exactly 30 seconds long and has long periods of silence and other artifacts. Not sure if it's an issue with their API or something with our preprocessing:

<img width="1076" alt="image" src="https://github.com/user-attachments/assets/a9895d57-8ac8-40f6-939d-f0aaf3de96bd" />

Reply:
I think it's the Fal API. The zero-gpu demo in their org is a lot better.